compiler/rustc_lint/src/context/diagnostics.rs

```diff
@@ -261,6 +269,8 @@ pub(super) fn builtin(
                 }
 
                 is_feature_cfg |= best_match == sym::feature;
+            } else if let Some(cfg_name) = single_possibility {
+                db.help(format!("try `#[cfg({} = \"{}\")]`", cfg_name, name));
```

Comment:
This should be an applicable suggestion. Look above for examples, but maybe something like this.
```suggestion
                db.span_suggestion(
                    name_span.to(value_span),
                    "found config with similar value",
                    format!("{cfg_name} = \"{name}\""),
                    Applicability::MaybeIncorrect,
                );
```